### PR TITLE
Fix support for optional parameters in setters

### LINKED ISSUE #24895: Setter optional parameter breaks in 3.1
**Godot version:**  3.1.beta1.official (x11.64)

**OS/device including version:** Linux Fedora 29 x86_64

**Issue description:**
Using default parameters on setters now breaks in godot 3.1, giving the error _"Parse Error: Setter function needs to receive exactly 1 argument."_

**Steps to reproduce:**
The following works on 3.0, but not in 3.1:

```
var state = "IDLE" setget _set_state

func _set_state(value, force=false):
    ...
```
As can be seen in the reproduction script above, it works on 3.0 but breaks in 3.1 with the following output:
```
SCRIPT ERROR: GDScript::reload: Parse Error: Setter function needs to receive exactly 1 argument. See '_set_state()' definition at line 5.
   At: res://setter_issue.gd:3.
ERROR: reload: Method/Function Failed, returning: ERR_PARSE_ERROR
   At: modules/gdscript/gdscript.cpp:580.
```
It's seems to be due this commit: 743053734f187c220250d88e4e475b7a87767cbc
on this part: https://github.com/godotengine/godot/blob/b16c309f82c77d606472c3c721a1857e323a09e7/modules/gdscript/gdscript_parser.cpp#L7421

**Minimal reproduction project:**
Use this with `godot -s setter_issue.gd`.  Here is the content of the file:
```
# setter_issue.gd
extends SceneTree

var state = "IDLE" setget _set_state

func _set_state(value, force=false):
	print("set state:", value, force)
	state = value

func _init():
    print("Hello!")
    quit()
```

[setter_issue.gd.zip](https://github.com/godotengine/godot/files/2747185/setter_issue.gd.zip)

### PR DESCRIPTION
**Pull request by AllanDaemon**
Don't assume that the only types of arguments are mandatory (without default).

*Bugsquad edit:* Fixes #24895.

### CODE REVIEW

**Comment by AllanDaemon** [comment]
This is a simple solution to #24895. It would be good to check if there are other cases with similar problems in this file.
Also, I created a helper function, but I don't think this is the best name for it, but it's the only thing I could think now.

**Comment by bojidar-bg** [comment]
Ah, also, I think this PR might cause some problems with this (otherwise-valid) code:
```gdscript
export var amount = 5 setget set_amount

func _ready():
    set_amount() # Ensure child nodes are properly updated

func set_amount(new_amount = amount):
    amount = new_amount
    if is_inside_tree():
        $Label.text = "Amount: %d" % amount
```

**Comment by akien-mga** [comment]
Please be more explicit about the bug being fixed (in plain English) in commit logs, a reference to the GitHub issue is good but not enough. Something like this would be better:
```
Fix support for optional parameters in setters

Fixes #24895.
```

**Comment by akien-mga** [comment]
CC @vnen

**Comment by vnen** [comment]
Well, I agree with @bojidar-bg's comments. Waiting for an answer from the PR author.

**Comment by AllanDaemon** [comment]
I fixed the points made by @bojidar-bg . 

I'm using the following code to test (using `godot -s setter_issue.gd`). I don't know if this could be incorporated into the test codebase. I've looked around and didn't find a place it would fit.
```
# setter_issue.gd
extends SceneTree

export var amount = 5 setget set_amount, get_amount

var state = "IDLE" setget _set_state, _get_state

func _set_state(value, force=false):
	printt("set state:", value, force)
	state = value

func _get_state():
	return state

func set_amount(new_amount = amount):
	amount = new_amount
	printt("Amount:", amount)
	# if is_inside_tree():
		# $Label.text = "Amount: %d" % amount

func get_amount(doubled=false):
	if doubled:
		return amount * 2
	else:
		return amount

func _ready():
	set_amount() # Ensure child nodes are properly updated
	prints("Amount calls:", self.amount, get_amount(), get_amount(true))

func _init():
	print("Hello!")
	_ready()
	quit()
```

**Comment by akien-mga** [comment]
Could you squash the commits together? There are more commits than lines changed ;)

**Comment by AllanDaemon** [comment]
> Could you squash the commits together? There are more commits than lines changed ;)

Done. Is there anything else that I should do?

**Comment by AllanDaemon** [comment]
Will this fix make to 3.1 release?

**Review by bojidar-bg — APPROVED** [review]
LGTM

**Comment by vnen** [comment]
Seems okay to me as well. Thanks!
